security/pkg/pki/ca/ca.go

```diff
@@ -329,8 +331,14 @@ func updateCertInConfigmap(namespace string, client corev1.CoreV1Interface, cert
 // GenKeyCert() generates a certificate signed by the CA and
 // returns the certificate chain and the private key.
 func (ca *IstioCA) GenKeyCert(hostnames []string, certTTL time.Duration) ([]byte, []byte, error) {
-	opts := util.CertOptions{
-		RSAKeySize: 2048,
+	opts := util.CertOptions{}
+
+	_, privKey, _, _ := ca.keyCertBundle.GetAll()
+	switch (*privKey).(type) {
```

Comment:
Pretty sure RSA keys can sign ECC, and the other way around.

Reply:
Yes that’s correct.

I have only seen in odd cases where ECC certs signed by RSA might get rejected.